### PR TITLE
Replace deprecated display apis with windowmetrics apis

### LINKED ISSUE #16073: Android 12: User experience - Replace deprecated Display APIs with WindowMetrics APIs
Parent #16062

[Display#getRealSize and getRealMetrics: deprecation](https://developer.android.com/about/versions/12/behavior-changes-all#displaymetrics)

`WindowMetrics` API was introduced in `Android 11` and these methods were deprecated:

`Display.getSize()`
`Display.getMetrics()`

In `Android 12`, following additional methods are deprecated:

`Display.getRealSize()`
`Display.getRealMetrics()`

It is recommended to use `WindowMetrics` instead of these deprecated methods.

Also, consider replacing deprecated methods from `WordPress-Utils-Android`'s `DisplayUtils` and `ImageUtils` wherever applicable.

### PR DESCRIPTION
**Pull request by AjeshRPai**
Parent #16073

As part of the changes for Android 12. This PR replaces the usages of the deprecated functions in `DisplayUtils.kt` in the WordPress Utils library. 

Functions replaced in the PR are 
1. `DisplayUtils.getDisplayPixelWidth(context)`with `DisplayUtils.getWindowPixelWidth(context)`
2. `DisplayUtils.getDisplayPixelHeight(context)` with  `DisplayUtils.getWindowPixelHeight(context)`
3. `DisplayUtils.getDisplayPixelSize(context) `with `DisplayUtils.getWindowSize(context)`

To test:

Testing instructions

The APIs are used in the following places

1. `MediaBrowserActivity.java`
2. `MediaGridAdapter.java`
3. `MediaPreviewFragment.java`
4. `MediaSettingsActivity.java`
5. `MediaSettingsActivity.java`
6. `EditPostActivity.java`
7. `PostListFragment.kt`
8. `EditorPhotoPicker.kt`
9. `ReaderPhotoViewerFragment.java`
10. `ReaderPostListFragment.java`
11. `ReaderResourceVars.java`
12. `ReaderCommentAdapter.java`
13. `ReaderPostAdapter.java`
14. `ReaderPostTagsUiStateBuilder.kt`
15. `HistoryDetailContainerFragment`
16. `EditorMediaUtils.java`
17. `StockMediaPickerActivity.java`
18. `ReaderThumbnailStrip.java`

In order to confirm that there is no regression issue we can test the following places. 

### Test 1 - Test on Android 11 and Android 12
`ReaderPhotoViewerFragment.java`

- Log in to the app 
- Go to the reader tab 
- Click on any reader post with multiple images 
- Click at the image and swipe between the images 
- Click back 
- Switch off the WIFI/Internet 
- Click at the image and swipe between the images 
- Verify that the Image is shown properly in portrait and landscape orientation 

### Test 2
`EditorMediaUtils.java`	

- Log in to the app 
- Go to the My Site Dashboard 
- Create a post with a video 
- Verify that the Video thumbnail is shown properly 

### Test 3
`ReaderCommentAdapter.java`

- Log in to the app 
- Go to a post with comments 
- Verify that the comments and nested comments(replies to the comments) are shown with the correct padding 

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:
- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by AjeshRPai** [comment]
cc:@hichamboushaba 👋🏼 
I have replaced the deprecated Display metrics API with Window metrics API in this PR. As the deprecated APIs are no longer needed in WordPress Utils, We can remove them from the repo.

File: libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/EditorMediaUtils.java [hunk 1]
```diff
@@ -44,10 +44,10 @@ public static BitmapDrawable getAztecPlaceholderDrawableFromResID(Context contex
     }
 
     public static int getMaximumThumbnailSizeForEditor(Context context) {
-        Point size = DisplayUtils.getDisplayPixelSize(context);
-        int screenWidth = size.x;
-        int screenHeight = size.y;
-        int maximumThumbnailWidthForEditor = screenWidth > screenHeight ? screenHeight : screenWidth;
+        Rect size = DisplayUtils.getWindowSize(context);
+        int screenWidth = size.height();
+        int screenHeight = size.width();
```
Comment (ashiagr) [inline libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/EditorMediaUtils.java]: Can we recheck the assignments: `size.height()` to `screenWidth` and `size.width()` to `screenHeight`?

File: WordPress/src/main/java/org/wordpress/android/util/DisplayUtilsWrapper.kt [hunk 2]
```diff
@@ -8,14 +8,15 @@ import javax.inject.Inject
...
+    fun isLandscapeBySize() =
+            getDisplayPixelWidth() > DisplayUtils.getWindowPixelHeight(contextProvider.getContext())
 
     fun isLandscape() = DisplayUtils.isLandscape(contextProvider.getContext())
 
     fun isTablet() = DisplayUtils.isTablet(contextProvider.getContext()) ||
             DisplayUtils.isXLargeTablet(contextProvider.getContext())
 
-    fun getDisplayPixelHeight() = DisplayUtils.getDisplayPixelHeight(contextProvider.getContext())
+    fun getDisplayPixelHeight() = DisplayUtils.getWindowPixelHeight(contextProvider.getContext())
```
Comment (ashiagr) [inline WordPress/src/main/java/org/wordpress/android/util/DisplayUtilsWrapper.kt]: Can we rename `getDisplayPixelHeight()` to `getWindowPixelHeight()`?

**Review by ashiagr — COMMENTED** [review]
Changes look good overall, I left two questions for you to consider. 
Also, we might want to create a PR in `WordPress-Utils` to remove deprecated methods. Let me know if you need help with it. You can add me as a reviewer.

File: libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/EditorMediaUtils.java (hunk 1 above)
Comment (AjeshRPai) [reply libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/EditorMediaUtils.java]: @ashiagr Oops, I missed that, Done in [03f1d52](https://github.com/wordpress-mobile/WordPress-Android/pull/16677/commits/03f1d52899bf75d168b4e911e28c81e62e4a3221)

File: WordPress/src/main/java/org/wordpress/android/util/DisplayUtilsWrapper.kt (hunk 2 above)
Comment (AjeshRPai) [reply WordPress/src/main/java/org/wordpress/android/util/DisplayUtilsWrapper.kt]: Yup right. Done in [b9684d4](https://github.com/wordpress-mobile/WordPress-Android/pull/16677/commits/b9684d4ac4887bde4c7b93bd03dccf9544ff5fa9)

**Comment by AjeshRPai** [comment]
> Also, we might want to create a PR in WordPress-Utils to remove deprecated methods. Let me know if you need help with it. You can add me as a reviewer.

Yup, Thanks. I have created a PR for the same [here](https://github.com/wordpress-mobile/WordPress-Utils-Android/pull/106)

**Review by ashiagr — APPROVED** [review]
LGTM 👍 

:shipit:

**Pull request merged**